src/drivers/angular2.js

```diff
@@ -0,0 +1,99 @@
+        {
+            Component: window.ng.core.Component,
+            NgModule: window.ng.core.NgModule,
+            ElementRef: window.ng.core.ElementRef,
+            NgZone: window.ng.core.NgZone,
+            BrowserModule: window.ng.platformBrowser.BrowserModule
+        } 
+        : false;
+    },
+
+    register(xcomponent, { Component, NgModule, ElementRef, NgZone, BrowserModule }) {
+
+        xcomponent.log('initializing angular2 component');
+        
+        const getBindingMetadata = () => {
+            const inputs = [];
+            if (xcomponent.looseProps) {
+                inputs.push('props');
+            } else {
+                for (let key of Object.keys(xcomponent.props)) {
+                    inputs.push(key);
+                }
+            }
+            return { inputs };
+        };
+
+
+        function getProps(component) {
+            const props = component.props || omit(component, ['$xContext']);
+            return replaceObject(props, (value, key, fullKey) => {
+                if (typeof value === 'function') {
+                    return function () {
+                        let result;
+                        component.$xContext.zone.run(() => {
+                            result = value.apply(this, arguments);
+                        });
+                        return result;
+                    };
+                }
+            });
+        }
+
+        const bindingMetadata = getBindingMetadata();
+
+        const Angular2Component = 
+            Component({
+                selector: xcomponent.tag,
+                template: `
+                    <div></div>
+                `,
+                inputs: bindingMetadata.inputs
+            })
+            .Class({
+                constructor: [ElementRef, NgZone, function(elementRef, zone) { 
+                    this.$xContext = {
+                        elementRef,
+                        zone
+                    };
+                }],
+                ngOnInit: function () {
+                    const targetElement = this.$xContext.elementRef.nativeElement;
+                    const parent = xcomponent.init(getProps(this), null, targetElement);
+                    parent.render(targetElement);
+                    this.$xContext.parent = parent;
+                },
+                ngOnChanges: function() {
+                    if (this.$xContext.parent) {
+                        this.$xContext.parent.updateProps(getProps(this));
+                    }
+                }
+            });
+
+
+        const Angular2Module = NgModule({
+            imports: [BrowserModule],
+            declarations: [Angular2Component],
+            exports: [Angular2Component]
+        })
+        .Class({
+            constructor: function () {
+                
+            }
+        });
+        
+        xcomponent.driverResults.angular2 = {
+            module: Angular2Module
+        };
+
+        return xcomponent;
```

Comment:
First I don't think we have to import ```ng.platformBrowser.BrowserModule``` in our wrapping module, I will check that out. 
Second, I still feel like we should be passing a module instead of a component. As in future we may need to add a service or a pipe .etc and only passing a component won't help. Users in angular world used to share plugins and components via modules due to the same reason. As an example: https://material.angular.io/guide/getting-started where each UI component is imported via importing its module.